### ISSUE TITLE
建议RKLLM  添加约束解码，方便function call 的调用

### ISSUE THREAD

**Issue by maxtrisun** [airockchip/rknn-llm#528]
目前RKLLM 不支持外挂 grammar / JSON-schema / regex 约束解码，也没有 xgrammar/outlines/lm-format-enforcer 式的逐步 logits-processor 钩子。 希望RKLLM 可以支持这些内容， 使RKLLMk可以像llama.cpp 的 GBNF grammar一样，将格式错误降低

**Comment by yg1031** [comment]
请问有什么好办法吗?

**Comment by maxtrisun** [comment]
大致有3种方法吧: 

#### 1. 最理想的方式是: RKLLM也可以像llama.cpp一样支持 GBNF grammar等约束解码，这样在底层支持，给顶层使用者带来很多便利，感觉这种方式也可以提升NPU的利用率；

#### 2. 在目前这种情况下(RKLLM 不支持约束解码)，若更在意json格式准确率，好像可以直接使用llama.cpp，利用它的约束解码，缺点是利用不了NPU了，所有计算都走CPU了，可能造成算力紧张，tok/s 下降； 

#### 3. 在生成后添加一个schema 校验， 生成的格式不对，失败重采样，这是一种事后采样的方式，没有格式硬保证。

#### 最期望的还是方法RKLLM 底层提供格式约束，形成格式硬约束保证，这样在function call 场景中最方便了。
